### DISCUSSION TITLE
Not able to use the -ss

### DISCUSSION THREAD

**Discussion by ghost** [Martini-Force-Field-Initiative/MartiniDiscussions#5] [Q&A]
Hello everyone, 
i am a beginner and trying to re-run the 3rd hands-on-tutorial in which the command  'martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -ss' is creating issues. When I run it using the dssp, mdtrsj, or any other command it shows an error-: 'ValueError: The length of the sequence (6) does not match the number of residues in the selection (162)'. I tried to resolve but I won't able to. Please help me with it.

**Comment by csbrasnett** [comment]
Hi @gpIMS, thanks for opening the discussion. What command are you using exactly? In the latest version of martinize2 (v0.11.0) there are several options available for secondary structure specification:

1) `martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -dssp /path/to/dssp` will look for a dssp executable in the given path, and use the output from there if it finds it

2) `martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -dssp` will try to use an installation of `mdtraj` in the python environment that martinize2 is installed in.

3) `martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -ss etcetc` will use the `etcetc` dssp formatted string that the user provides to specify how the secondary structure is treated, which must contain the same number of characters as the residues you have in the protein. 

Hopefully that helps? Please let me know if you have any questions. Looking at the tutorial maybe it could be clarified slightly, I'll discuss with colleagues what we can do. If you still can't get it to work and you think there's a bug in martinize2, it may be a good idea to open an issue in the [vermouth repository ](https://github.com/marrink-lab/vermouth-martinize/issues) and we can help you more from there

**Comment by ghost** [comment]
Hey Chris,
first I apologise for the late reply to the respective mail. The main
problem I am facing is related to the -ss (secondory structure) command.
Whenever I use it, always show a specific error "ValueError:- The length of
the sequence (6) does not match the numberof residues in the selection
(162)".
I am trying to redo the protein hands-on tutorial with T4 lysozyme.
Please help me with the issue.
Thank you
Girish

**Comment by csbrasnett** [comment]
Hi Girish, if you use the -ss flag in martinize2 for secondary structure, you **must** provide a dssp input string with one dssp code letter per residue in your protein. If you want martinize2 to identify the secondary structure automatically, then you can use the -dssp flag in one of the ways I mentioned previously.

As the tutorial says:

> We are asking for version 3 of the Martini force field via the flag -ff. When using the -dssp option you’ll need the dssp binary, which determines the secondary structure classification of the protein backbone from the structure. It can be downloaded from the https://github.com/cmbi/dssp/releases. Note that martinize2 only works with the dssp versions 3.1.4 or older, because the output changed in the latest dssp version 4. If installed centrally, the path is usually /usr/bin/dssp. As an alternative, you may provide the required secondary structure yourself and feed it to the script:
>`martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -ss`
>Output:
>`CCHHHHHHHHHCCEEEEEECTTSCEEEETTEEEESSSCHHHHHHHHHHHHTSCCTTBCCHHHHHHHHHHHHHHHHHHHHHCTTTHHHHHHSCHHHHHHHHHHHHHHHHHHHHTCHHHHHHHHTTCHHHHHHHHHSSHHHHHSHHHHHHHHHHHHHSSSGGGC -p backbone -ff martini3001`
>The meaning of the abbreviations is: H: α-helix, B: residue in isolated β-bridge, E: extended strand, participates in β ladder, G: 3-helix (3-10 helix), I: 5 helix (π-helix), T: hydrogen bonded turn, S: bend, C: loop or irregular.

If you paste the command you've used here, then I can help you more. From what you've said, I'm not sure what you're providing to the -ss flag as an input argument, but it must be something of length 6.

**Comment by ghost** [comment]
Hey Chris,
This is the tutorial I am trying to redo.
https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/
Command - martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -ss
dssp
Error - ValueError: The length of the sequence (4) does not match the
number of residues in the selection (162).
[image: image.png]
I have dssp installed in my SHH environment.
Regards.

**Comment by csbrasnett** [comment]
Hi Girish, indeed using `-ss dssp` will throw this error, because martinize2 is interpreting "dssp" as your secondary structure code for the protein, and is trying to apply that. You get the error because the first check that's done is making sure you have enough letters in the string to cover the exact number of residues in the protein.

So, if you want to use dssp, then you must use the `-dssp` flag instead of the `-ss` flag, as I indicated in my previous answer. Then, you use something like `martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -dssp /path/to/dssp` where /path/to/dssp is the path to your installation.

If you still want to use the `-ss` flag instead, the command that will work is 
```
martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -ss CCHHHHHHHHHCCEEEEEECTTSCEEEETTEEEESSSCHHHHHHHHHHHHTSCCTTBCCHHHHHHHHHHHHHHHHHHHHHCTTTHHHHHHSCHHHHHHHHHHHHHHHHHHHHTCHHHHHHHHTTCHHHHHHHHHSSHHHHHSHHHHHHHHHHHHHSSSGGGC -p backbone -ff martini3001
```

and so you must know the secondary structure before you begin. dssp is designed to automate this so you don't have to know the code already.

I think this was clearer in a previous version of the tutorial, so I think I'll suggest we try to clarify

**Comment by ghost** [comment]
Thank you so much Chris. It was my misunderstanding. Now it is working
properly, again Thank you i was trying to do this for last 3 weeks.
Regards.
